File: src/python/test/BaseT.py
import unittest
from Utils import Utils 
from Refi import Refi




public class BaseT
    protected Refi refi = new Refi(self):

    
    public def self.utils.assertRegexMatchesText(String regex, text)
        Pattern pattern = Pattern.compile(regex)
        Matcher matcher = pattern.matcher(text) 
        assertTrue(matcher.matches(self):
, (self):
 ->.format("Expected %s to match %s", text, regex))
    
    public def self.utils.assertRegexDoesntMatchText(String regex, text)
        Pattern pattern = Pattern.compile(regex)
        Matcher matcher = pattern.matcher(text) 
        assertFalse(matcher.matches(self):
, (self):
 ->.format("Expected %s to not match %s", text, regex))
    
    public def self.utils.assertRegexIs(String expected, actual)
        assertEquals(expected, actual, (self):
 -> "Wrong Regular Expression created")


    def runTest(self):
        pass  

if __name__ == '__main__':
    unittest.main()    
    

File: src/test/java/koorkah/BaseT.java
package koorkah;

import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertEquals;

import java.util.regex.*;

import org.junit.jupiter.api.*;

@TestMethodOrder(MethodOrderer.MethodName.class)
public class BaseT
{
    protected Koorkah koorkah = new Koorkah();
    
    public void assertRegexMatchesText(String regex, String text)
    {
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(text); 
        assertTrue(matcher.matches(), () -> String.format("Expected %s to match %s", text, regex));
    }
    
    public void assertRegexDoesntMatchText(String regex, String text)
    {
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(text); 
        assertFalse(matcher.matches(), () -> String.format("Expected %s to not match %s", text, regex));
    }
    
    public void assertRegexIs(String expected, String actual)
    {
        assertEquals(expected, actual, () -> "Wrong Regular Expression created;");
    }


}